### ISSUE TITLE
Footnotes not woking for mobi 

### ISSUE THREAD

**Issue by coiby** [tuxu/sicp-epub-svgmath#1]
Hi

Thank you for providing the mobi version. 

Everything is perfect except for the footnote. If I try to open the footnotes on Kindle Paperwhite, it just goes to next page directly. I also noticed when reading it using Calibre on PC, most of the footnotes are shown properly but footnotes like #71 are blank.

**Comment by tuxu** [comment]
Thanks for the report! It seems like there are very subtle differences on how one should create footnotes in Epub/Mobi. With the few changes in tuxu/sicp@796fbe1, the footnotes appear on my Paperwhite. They also work in Calibre now.

**Comment by coiby** [comment]
Thank you for the fix!

The footnotes can be open correctly now. However I notice Paperwhite don't handle newlines like modern browsers. For example, the html for footnote#67 is
```html
The
ability to directly manipulate procedures provides an analogous increase in the
expressive power of a programming language.  For example, in 
<a href="1_002e3.xhtml#g_t1_002e3_002e1">1.3.1</a>
```
When it's rendered by Paperwhite, these newlines are also rendered, i.e. a new line will begin after "The", "increase in the", "For example, in".

It seems we need to re-format original HTML files to remove extra newlines.

**Comment by tuxu** [comment]
I just removed these extra newlines in tuxu/sicp@fb37363. Footnote rendering is still far from being perfect, though, but I suppose it's the best we can do at the moment.

Few remaining issues I noticed: 

- Footnotes with equations/symbols stop rendering on the first occurrence of the equation.
- Footnotes with code in &lt;pre&gt; tags is not rendered with fixed width or styled by CSS.
- Links from the footnote section back into the main text end up somewhere else.

Luckily, the footnotes render properly in the footnote section, so one can jump there when a footnote is more demanding and return to the main text with the back button.

**Comment by coiby** [comment]
Good job! Thank you for all the efforts!

**Comment by coiby** [comment]
I noticed this fix may introduce another problem. Some equations are now shown as texts. For example, the caption of Figure 2.24 is now rendered as
```html
Representation of \( {3 + 4i} \) in rectangular form.
```
Maybe non-greedy regexes should be used?

**Comment by tuxu** [comment]
Oh, I see. The problem was actually not the greedy regex. The problem was that newlines were substituted by spaces also for LaTeX equations inside footnotes. When these contained comments, they failed at the rendering stage then and could no longer be replaced by the SVG versions.

This is fixed now in tuxu/sicp@cff71d8, and I'll also push fixed EPUB and MOBI files in a minute.

**Comment by coiby** [comment]
Thank you for the fix and also the elaboration!
